### DISCUSSION TITLE
DDL to TF?

### DISCUSSION THREAD

**Discussion by gliptak** [Snowflake-Labs/terraform-provider-snowflake#685] [Q&A]
Many DB tools generate `DDL` commands like:

```
CREATE TABLE employees (
    id            INTEGER       PRIMARY KEY,
    first_name    VARCHAR(50)   not null,
);
```

the corresponding `TF` snippet might be: 

```
  column {
    name     = "id"
    type     = "INTEGER"
    nullable = false
  }

  column {
    name     = "first_name"
    type     = "VARCHAR(50)"
    nullable = false
  }

  primary_key {
    name = "my_key"
    keys = ["id"]
  }
```

What are the recommended ways to use(/convert?) `DDL` generated by various DB tools in Snowflake Terraform provider?

**Comment by Synohara** [comment]
I created a cli tool to convert DDL to TF using ChatGPT. It's not perfect, but it works pretty well.
https://gist.github.com/Synohara/78e246e8ab4e156a2b4513df4760067c
